ai/docs/GPT Online Model Fine-tuning Documentation.md

```diff
@@ -0,0 +1,354 @@
+
+```plain
+python3 finetune.py -k <openai_api_key> -d output.jsonl
+```
+Or:
+```plain
+python3 finetune.py --key <openai_api_key> --data output.jsonl
+```
+This is default training setting-up. For setting hyper-parameters (batch_size, learning_rate_multiplier, n_epochs) and validation file, please write the configuration file like this in "train.cfg". Please pay attention to the names of configurations, which should be matched precisely.
+```plain
+openai_api_key = <openai_api_key>
+training_file = <training_file>
+model = gpt-3.5-turbo
+n_epochs = 3
+batch_size = 1
+learning_rate_multiplier = 2
+validation_file = <validation_file>
+```
+Then run the command:
+```plain
+python3 finetune.py --config train.cfg
+```
+Or:
+```plain
+python3 finetune.py -c train.cfg
+```
+Once your fine-tuning job created, you can get the training status [on the website](https://platform.openai.com/finetune). For the original python API, please refer to [OpenAI API](https://platform.openai.com/docs/api-reference/fine-tuning/create).
+
+As for Linux Shell (with curl installed), you can run the 'curl' command to upload your file with environment variables set up. As for the simplest UI way, you can go [official website UI](https://platform.openai.com/finetune) to submit fine-tuning jobs, by clicking "Create new". Please refer to official website if you want to create fine-tuning jobs in these ways.
+
+### 3. Use a fine-tuned model
+To use a fine-tuned model, you can run the command below, and the model's answer will be returned in stdout. The fine-tuned model name can be acquired on [your OpenAI account](https://platform.openai.com/finetune/).
+
+```plain
+python3 infer.py --key <openai_api_key> --model <model_name> -q <your_question>
+```
+The model name can also be acquired using the fine-tuning job id, which will be outputed right after fine-tuning job created. So you could just input the fine-tuning job id to call the model.
+```plain
+ python3 infer.py --key <openai_api_key> --job <ft_job_id> -q <your_question>
+```
+To customize the prompt, you can write a configuration file for testing. Please pay attention to the names of configurations, which should be matched precisely. An example is like this in "infer.cfg":
+```plain
+openai_api_key = <openai_api_key>
+ft_job_id = <job_id>
+model = <model_name>
+max_tokens = <max_tokens>
+temperature = 0
+sys_prompt = <your_prompt>
+user_input = <your_question>
+```
+Then run the command below. The model's response will be returned.
+```plain
+python3 infer.py --config infer.cfg
+```
+
+## Code Explanation
+As for the "data_process.py" in Step 1, it matches the data format in the "Supposed" paragraph. It will first convert the input json into data_list, which is formatted, then dump it into the jsonl file. The dicts_to_jsonl function is implemented in "utils.py" within the same folder.
```

Comment:
If you only want to introduce the general function of the file, you can insert such descriptions into the respective paragraphs. Otherwise if you want to explain the specific purpose of the code, it's better to add descriptions to the code comments.

Reply:
Updated